lib/facter/custom_facts/core/execution/popen3.rb

```diff
@@ -20,7 +20,15 @@ def self.popen_rune(cmd, opts, child_io, parent_io) # :nodoc:
               return yield(*result)
             ensure
               parent_io.each(&:close)
-              Process.wait(pid)
+              begin
+                Process.wait(pid)
+              rescue Errno::ENOENT
+                # For some reason, the first Process.wait executed in JRuby
+                # always fails with ENOENT. However, the command output is
+                # filled in so we just need to silently continue.
+                # https://github.com/jruby/jruby/issues/5971
+                raise unless RUBY_PLATFORM == 'java'
```

Comment:
could we call micro-sleep and call Process.wait(pid) again instead on silently ignoring the error?

Reply:
Subsequent calls to `Process.wait(pid)` with the same pid fail with `ECHILD` so I assume the PID doesn't exist anymore.

Reply:
Can confirm without any sleep:
> Errno::ECHILD: No child processes - No child processes